src/modules/create-market/components/create-market-form/create-market-form.jsx

```diff
@@ -139,7 +139,8 @@ export default class CreateMarketForm extends Component {
 
     let value = rawValue
 
-    if (value !== '') {
+    const regExp = new RegExp('^[0-9]+\\.[0]{0,' + decimals + '}$')
+    if (value !== '' && !regExp.test(value)) {
```

Comment:
@JohnDanz The problem is that when the value entered in the textbox is "0.0", it was doing a `parseFloat` on that, which got evaluated as "0", so `value` was getting set to 0. Ultimately, this was resetting the value in the textbox to "0", so no values like "0.01" could be entered. 

I'd definitely prefer a simpler solution, but I'm not sure what you described above addresses this. When `value` is "0.0", I think doing `let value = parseFloat(value)` will still have the same problem.

Reply:
if value === 0 (after parsing) then setState to the rawValue, instead of value. This way we can simplify and then still allow for 0.000 being entered without being annoyingly refreshed to 0

Reply:
Ok, I see what you're saying, but this still would not allow the user to enter something like "1.05" in the textbox, because once they enter "1.0", the textbox will revert this to "1".  The other (smaller) issue is that they could enter an unlimited number of zeros, like "0.00000000000".  This isn't actually incorrect, but may seem kind of strange to the user since, in all other cases, they can only enter 2 decimal places.

Reply:
use the raw value. The text box should never correct the value inside it, it should display what the user puts in, but if it's invalid it should show a invalid error and not let you proceed. If it's rewriting 1.0 into 1, then that's a sign of it not working correctly. if we are limiting what a user can input to 2 decimal places, we shouldn't be. We should either round what they put in prior to sending it to 2 decimals, or show them error text and invalidate until they limit their input to 2 decimals if it's import to limit them.

Reply:
Ok, yeah, the code was removing 0s after the decimal point. I was hesitant to change a lot since this is used in multiple parts of the UI.  Also, the function itself accepts a `decimals` parameter, so the function is designed to limit a textbox to a certain number of decimal places.

Reply:
```
value = parseFloat(value)
value = parseFloat(value.toFixed(decimals))
```
we can keep the way you are doing it for now actually, as long as it works it's fine, we can revisit this stuff in peccary, but the above code snippet (which is actually below this comment in the actual file) needs to be looked at. `parseFloat` returns a JS Number and JS Numbers can't do `toFixed` so this looks like it would cause an error.

Reply:
@JohnDanz I believe calling `toFixed` on a JS Number should be ok. For example, here's a case where a `parseFloat` value is having `toFixed` called on it: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed.